egg/core/trainers.py

```diff
@@ -189,6 +189,12 @@ def train_epoch(self):
 
         self.optimizer.zero_grad()
 
+        try:
+            data_len = len(self.train_data)
+        except (TypeError, AttributeError):
+            # len not implemented
+            data_len = -1
+
```

Comment:
se previous comment, I don't think we need this here. Games having a callback could do that on their own